src/video_core/amdgpu/liverpool.h

```diff
@@ -889,10 +889,54 @@ struct Liverpool {
             return !info.linear_general;
         }
 
-        NumberFormat NumFormat() const {
+        [[nodiscard]] DataFormat DataFormat() const {
+            return RemapDataFormat(info.format);
+        }
+
+        [[nodiscard]] NumberFormat NumFormat() const {
             // There is a small difference between T# and CB number types, account for it.
-            return info.number_type == AmdGpu::NumberFormat::SnormNz ? AmdGpu::NumberFormat::Srgb
-                                                                     : info.number_type.Value();
+            return RemapNumberFormat(info.number_type == NumberFormat::SnormNz
+                                         ? NumberFormat::Srgb
+                                         : info.number_type.Value());
+        }
+
+        [[nodiscard]] CompMapping Swizzle() const {
+            // clang-format off
+            static constexpr std::array<std::array<CompMapping, 4>, 4> mrt_swizzles{{
```

Comment:
Reading the register reference, the single component case doesn't look correct to me, since it maps Red to zero
```
With one component, this
selects which colour channel to map to the single render target component
(STD: R=>0; ALT: G=>0; STD_REV: B=>0; ALT_REV: A=>0).
```
The existing component swap implementation also had this bug seems, so might be a good opportunity to fix now. PAL seems to have a handy function for determining order https://github.com/GPUOpen-Drivers/pal/blob/042362399cdac1019fbc7f0ace8489aee2907883/src/core/hw/gfxip/gfx6/gfx6FormatInfo.cpp#L284

Reply:
I'm not quite sure I understand what you mean, for single component case I have (R000), (0R00), (00R0), (000R) currently which as far as I can tell matches this?

Reply:
Seems I misread the docs, disregard this

Reply:
Or wait, one more question, I'm a bit confused. With a mapping of (0, R, 0, 0) assuming an input vector of (1, 2, 3, 4) would it result in (0, 1, 3, 4) ? If render target is R32, then we would write zero instead?

Reply:
Hmm I think currently I've defined and interpreted these backwards, setting a field to where it should go instead of setting where it should go to the source. Let me think on it a bit.